### DISCUSSION TITLE
How to open a new scratchpad in the current workspace?

### DISCUSSION THREAD

**Discussion by moshpirit** [i3/i3#6282] [Q&A]
I'm trying with this, but it doesn't seem to work:

```
bindsym $mod+Shift+Return exec --no-startup-id kitty --name jupyter-scratchpad --hold sh -c '~/.scripts/cowsay_pomodoro.sh' &
for_window [instance="jupyter-scratchpad"] floating enable, resize set 1400 905, border pixel 2, scratchpad, move position center, move scratchpad
for_window [class="konsole|kitty|Alacritty" floating] move to workspace current
```

**Comment by okraits** [comment]
You need to move an existing window into the scratchpad workspace and then you can show it.

Example for a keybinding:

`bindsym $mod+x move scratchpad; scratchpad show`

See also the userguide: https://i3wm.org/docs/userguide.html#_scratchpad

**Reply by moshpirit** [reply]
I can use scratchpads; I do have these keybinds on `$HOME/.config/i3/i3-configfiles/i3-keybinds.conf` (which is included)
```
bindsym $mod+minus scratchpad show
bindsym $mod+Shift+minus move scratchpad
```
The issue is that when I create a new one it doesn't get opened in the current workspace, but in where the terminals belong (ws1)

In my main file I have the include files along with the `set` variables (except for the modes):
```
# set the mod key to Super:
set $mod Mod4

set $ws1 "1 "
set $ws2 "2 "
set $ws3 "3 "
set $ws4 "4 "
set $ws5 "5 "
set $ws6 "6 "
set $ws7 "7 󰍩"
set $ws8 "8 󱄽"
set $ws9 "9 "
set $ws10 "10 "

# set scratchpads in the same workspace
set $current i3-msg -t get_workspaces | jq '.[] | select(.focused==true).name'

# Load the rest of the configuration files
include $HOME/.config/i3/i3-configfiles/i3-workspaces.conf
include $HOME/.config/i3/i3-configfiles/i3-plasmaconfig.conf
include $HOME/.config/i3/i3-configfiles/i3-colors.conf
include $HOME/.config/i3/i3-configfiles/i3-startup.conf
# include $HOME/.config/i3/i3-configfiles/i3-bar.conf
include $HOME/.config/i3/i3-configfiles/i3-windowsrules.conf
include $HOME/.config/i3/i3-configfiles/i3-modes.conf
include $HOME/.config/i3/i3-configfiles/i3-keybinds.conf
```
I tried including the 3 setences above in my main post just in case it was due to the inability to read external variables defined in other child files; but it doesn't seem to be the issue.

**Reply by moshpirit** [reply]
I've tried these alternatives too, but got me nowhere: 

### Option 1: Including the moving in the exec command:

``` shell
bindsym $mod+Shift+Return exec --no-startup-id \
    i3-msg "exec kitty --name jupyter-scratchpad --hold sh -c '~/.scripts/cowsay_pomodoro.sh'; \
    [instance="jupyter-scratchpad"] move to workspace $(i3-msg -t get_workspaces | jq -r '.[] | select(.focused).name')"
```

### Option 2: Using a bash script that does it and call it externally with a bindsym

```bash
#!/bin/bash
current=$(i3-msg -t get_workspaces | jq -r '.[] | select(.focused).name')
i3-msg "exec kitty --name jupyter-scratchpad --hold sh -c '~/.scripts/cowsay_pomodoro.sh'"
i3-msg "[instance='jupyter-scratchpad'] move to workspace $current"
echo current: $current
```

### Option 3: Waiting for the window to open and then move it

```bash
#!/bin/bash
current=$(i3-msg -t get_workspaces | jq -r '.[] | select(.focused).name')
i3-msg "exec kitty --name jupyter-scratchpad --hold sh -c '~/.scripts/cowsay_pomodoro.sh'"

while ! i3-msg -t get_tree | jq '.. | select(.name? == "jupyter-scratchpad")'; do
  sleep 0.1
done

i3-msg "[instance='jupyter-scratchpad'] move to workspace $current"
echo current: $current
```

In all scenarios the scratchpad is created in the same workspace (ws1); for bash scripts I get this error:

```
./.scripts/floating-scrachpad.sh 
[{"success":true}]
ERROR: Nothing to move: specified criteria don't match any window
[{"success":false,"error":"Nothing to move: specified criteria don't match any window"}]
current: 10 
```

The terminal gets open as a scratchpad and runs the command, but it doesn't get moved to the current workspace

Here is the `xprop` of the window:

```
_NET_WM_ICON(CARDINAL) = 	Icon (128 x 128):
	(not shown)

_NET_WM_DESKTOP(CARDINAL) = 0
_NET_WM_STATE(ATOM) = _NET_WM_STATE_FOCUSED
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
I3_FLOATING_WINDOW(CARDINAL) = 1
_NET_STARTUP_ID(UTF8_STRING) = "i3/kitty/975-34-archlinux_TIME13387926"
_VARIABLE_REFRESH(CARDINAL) = 1
_NET_WM_ICON_NAME(UTF8_STRING) = "xprop"
_NET_WM_NAME(UTF8_STRING) = "xprop"
WM_LOCALE_NAME(STRING) = "es_ES.UTF-8"
WM_CLIENT_MACHINE(STRING) = "archlinux"
WM_ICON_NAME(STRING) = "xprop"
WM_NAME(STRING) = "xprop"
XdndAware(ATOM) = BITMAP
WM_CLASS(STRING) = "jupyter-scratchpad", "kitty"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		window gravity: Static
WM_HINTS(WM_HINTS):
		Initial state is Normal State.
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_PID(CARDINAL) = 57063
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING
```

**Reply by moshpirit** [reply]
> You need to move an existing window into the scratchpad workspace and then you can show it.

Is there any way to automatically do that? I want to create a terminal as a floating scratchpad in the current workspace, but keep non scratchpads as their current behavior in the workspace 1.  Which is what I lack in the guide

**Comment by lbarsou** [comment]
Hello I can suggest you to check this add-on https://github.com/budlabs/i3ass/wiki/i3run , it might help you achieve what you want.
Maybe try that, i don't know if it's what you are seeking but:
bindsym $mod+Shift+Return exec --no-startup-id i3run --title "jupyter-scratchpad" -- kitty -T "jupyter-scratchpad" -e bash -c '~/.scripts/cowsay_pomodoro.sh'

**Reply by moshpirit** [reply]
Hi! Sorry, I didn't get notified with your comment! It doesn't seem to do anything. I tried running `i3run --title "jupyter-scratchpad" -- kitty -T "jupyter-scratchpad" -e bash -c '~/.scripts/cowsay_pomodoro.sh'`  at a terminal and I got a minimized scrachpad that instead of showing the result of the script (showing a cowsay with a pomodoro timer), it prints `104910981928880`. I tried changing it for an 'echo hi' and the result is the same but with a different number: `104910982025648`
